CharlieBackend.Api/Controllers/HomeworkStudentController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using CharlieBackend.Core.DTO.HomeworkStudent;
+using CharlieBackend.Business.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using CharlieBackend.Core;
+
+namespace CharlieBackend.Api.Controllers
+{
+    /// <summary>
+    /// Controller to make operations with homework from student
+    /// </summary>
+    [Route("api/homeworkstudent")]
+    [ApiController]
+
+    public class HomeworkStudentController : ControllerBase
+    {
+
+        private readonly IHomeworkStudentService _homeworkStudentService;
+
+        /// <summary>
+        /// HomeworkStudent controllers constructor
+        /// </summary>
+        public HomeworkStudentController(IHomeworkStudentService homeworkStudentService)
+        {
+            _homeworkStudentService = homeworkStudentService;
+        }
+
+        /// <summary>
+        /// Adds homework from student
+        /// </summary>
+        /// <param name="request">
+        /// 1.HomeworkId -  it is id homework which one student done 
+        /// 2.HomeworkTest - here Student can write his Homework or something else
+        /// 3.Attacment(LIst) -   here must be Id of Attachment where student uploaded materials for his Homework</param>
+        [SwaggerResponse(200, type: typeof(HomeworkStudentDto))]
+        [Authorize(Roles = "Student")] //?? admin 
```

Comment:
```suggestion
        [Authorize(Roles = "Student")]
```
Judging by this line's comment you're not sure if admin should have access to these endpoints or not.
But in your implementation of `CreateHomeworkFromStudentAsync()` you are using `_currentUserService.AccountId`, which no one but `Student` would have a valid value for.

So I think, despite the fact that admin can access all other endpoints (see the [endpoints' accessibility](https://github.com/ita-social-projects/WhatBackend/wiki/Endpoint-accessibility) page), we can leave it as is.